### PR TITLE
Add block resize test in nested ENV

### PR DESCRIPTION
**Pull request by qingwangrh**
ID:1955479

Signed-off-by: qingwangrh <qinwang@redhat.com>

### CODE REVIEW

**Comment by qingwangrh** [comment]
python ConfigTest.py --testcase=ansible_test.nested_block_resize --guestname=RHEL.8.4.0 --clone=no --driveformat=virtio_scsi
(1/1) Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.ansible_test.nested_block_resize: PASS (169.41 s)

**Comment by qingwangrh** [comment]
@YongxueHong @qcheng-redhat Please help to review.

**Comment by PaulYuuu** [comment]
Message queue is not used to connect to QMP/HMP.

**Comment by qingwangrh** [comment]
python ConfigTest.py --testcase=ansible_test.nested_block_resize --guestname=RHEL.8.4.0 --clone=no --driveformat=virtio_scsi --customsparams='toplevel_playbook = "test.yml"\nansible_extra_vars_fixed={"suite_type": "kar", "kar_args": "--testcase=nested_block_resize_l1 --guestname=RHEL.8.4.0 --machine=q35 --clone=no", "kar_customsparams": "mq_publisher = {{ hostvars.localhost.host_ip }}\u000amq_port = {{ mq_port }}" }'

(1/1) Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.ansible_test.nested_block_resize: PASS (173.56 s)

File: qemu/tests/nested_block_resize.py [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+    """
+    QEMU nested block resize test
+
+    1) Boot the main vm as L1, attach scsi data disk
+    2) Pass-through L1 data disk to L2
+    3) Run io on the data disk in L2.
+    4) Execute block_resize for data disk image on host.
+    5) Check L1 status should keep running.
+    6) Check L2 status should keep running.
+    7)
```
Comment (qcheng-redhat) [inline qemu/tests/nested_block_resize.py:26]: Is there any thing to do with step 7 ?

Comment (qingwangrh) [reply qemu/tests/nested_block_resize.py]: updated

File: qemu/tests/nested_block_resize_l1.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+# This decorator makes the test function aware of context strings
+@error_context.context_aware
+def run(test, params, env):
+    """
+    QEMU nested block resize test of L1
+
+    1) Boot the vm attach the disk from L1
+    2) Run io on the data disk then notify L1
+    3) Execute block_resize for data disk image on host.
+    4) L1 do block_resize and check status of L1 and L2.
```
Comment (qcheng-redhat) [inline qemu/tests/nested_block_resize_l1.py:17]: This step is executed on host, and already included in qemu/tests/nested_block_resize.py.

Comment (qingwangrh) [reply qemu/tests/nested_block_resize_l1.py]: updated

**Comment by qcheng-redhat** [comment]
python3 ConfigTest.py --testcase=ansible_test.nested_block_resize --guestname=RHEL.8.5.0 --clone=no --driveformat=virtio_scsi --customsparams='toplevel_playbook = "test.yml"\nansible_extra_vars_fixed={"suite_type": "kar", "kar_args": "--testcase=nested_block_resize_l1 --guestname=RHEL.8.5.0 --machine=q35 --clone=no", "kar_customsparams": "mq_publisher = {{ hostvars.localhost.host_ip }}\u000amq_port = {{ mq_port }}" }'

(1/1) Host_RHEL.m8.u5.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.5.0.x86_64.io-github-autotest-qemu.ansible_test.nested_block_resize: PASS (171.79 s)

LGTM

**Review by YongxueHong — APPROVED** [review]

File: qemu/tests/cfg/ansible_test.cfg [hunk 3]
```diff
@@ -15,3 +15,16 @@
             type = ansible_with_responsive_migration
             wait_response_timeout = 1800
             #mq_listen_port =
+        - nested_block_resize:
```
Comment (PaulYuuu) [inline qemu/tests/cfg/ansible_test.cfg:18]: Please remove this part in a single cfg file, ansible_test is not only used for nested testing.

Comment (qingwangrh) [reply qemu/tests/cfg/ansible_test.cfg]: Updated

File: qemu/tests/cfg/nested_block_resize.cfg [hunk 4]
```diff
@@ -0,0 +1,23 @@
+- ansible_test:
+    type = nested_block_resize
+    virt_test_type = qemu
+    playbook_repo = "https://github.com/ansible/test-playbooks.git"
```
[the remark this answers is not in the document]
Comment (qingwangrh) [reply qemu/tests/cfg/nested_block_resize.cfg:4]: It is not recommended to run single case due to many dependence like as nested environment need to deployed in advance.
The final running place is called in nested test loop.

File: qemu/tests/cfg/nested_block_resize.cfg [hunk 5]
```diff
@@ -0,0 +1,23 @@
+- ansible_test:
```
[the remark this answers is not in the document]
Comment (qingwangrh) [reply qemu/tests/cfg/nested_block_resize.cfg:1]: updated

**Comment by qingwangrh** [comment]
python ConfigTest.py --testcase=nested_block_resize --guestname=RHEL.8.4.0 --clone=no --driveformat=virtio_scsi --customsparams='toplevel_playbook = "test.yml"\nansible_extra_vars_fixed={"suite_type": "kar", "kar_args": "--testcase=nested_block_resize_l1 --guestname=RHEL.8.4.0 --machine=q35 --clone=no", "kar_customsparams": "mq_publisher = {{ hostvars.localhost.host_ip }}\u000amq_port = {{ mq_port }}" }'

(1/1) Host_RHEL.m8.u5.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.nested_block_resize: PASS (175.14 s)

**Comment by qingwangrh** [comment]
https://gitlab.cee.redhat.com/kvm-qe/kar/-/merge_requests/654

**Comment by qingwangrh** [comment]
https://gitlab.cee.redhat.com/kvm-qe/kar/-/merge_requests/654

File: qemu/tests/cfg/nested_block_resize.cfg [hunk 6]
```diff
@@ -0,0 +1,19 @@
+- nested_block_resize:
+    type = nested_block_resize
+    virt_test_type = qemu
+    toplevel_playbook = "site.yml"
+    playbook_timeout = 600
+    ansible_callback_plugin = debug
+    ansible_ssh_extra_args = "-o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null"
+    only virtio_scsi
+    type = nested_block_resize
```
Comment (qcheng-redhat) [inline qemu/tests/cfg/nested_block_resize.cfg:9]: Dup line.

Comment (qingwangrh) [reply qemu/tests/cfg/nested_block_resize.cfg]: updated

**Review by PaulYuuu — APPROVED** [review]

**Pull request merged**